src/chapters/more-on-functions/recursion-walkthrough2.rst

```diff
@@ -0,0 +1,133 @@
+Making A Function Call Itself
+==============================
+
+Congratulations! Identifying the base case is often the trickiest part of
+building a recursive function.
+
+We've made it this far with ``combineEntries``:
+
+.. sourcecode:: js
+
+   function combineEntries(arrayName){
+      if (arrayName.length === 1){
+         return arrayName[0];
+      } else {
+         //solve next small step
+         //call combineEntries again
+      }
+   }
+
+Now we are ready to take the next step.
+
+A Visual Representation
+------------------------
+
+For arrays with more than one entry, we need to make ``combineEntries``
+consider the first element and then *check what is left in the array*. If the
+rest of the array contains more than one item, ``combineEntries`` calls iteslf
+again and repeats the process with a smaller set of entries.
```

Comment:
These two sentences are a little ambiguous. One can read "check what is left" and "if the rest of the array contains more than one" to apply only to an array with two or more elements. It can also be read so as to apply only to an array with three or more elements ("rest of" could be what is left after taking the first element off). In fact, the recursive call also happens when there is only one element in the array.

Reply:
Fixed.